### PR TITLE
Remove upper bound on pygam

### PR DESCRIPTION
**Pull request by kklein**
Checklist
* [ ] Used a [personal fork of the feedstock to propose changes](https://conda-forge.org/docs/maintainer/updating_pkgs.html#forking-and-pull-requests)
* [ ] Bumped the build number (if the version is unchanged)
* [ ] Reset the build number to `0` (if the version changed)
* [ ] [Re-rendered]( https://conda-forge.org/docs/maintainer/updating_pkgs.html#rerendering-feedstocks ) with the latest `conda-smithy` (Use the phrase <code>@<space/>conda-forge-admin, please rerender</code> in a comment in this PR for automated rerendering)
* [ ] Ensured the license file is being packaged.

### CODE REVIEW

File: recipe/meta.yaml [hunk 1]
```diff
@@ -42,6 +42,7 @@ requirements:
     - matplotlib-base
     - pandas >=0.24.1,<1.4.0
     - scikit-learn >=0.22
+    - numpy >=1.18.5
```
Comment (pavelzw) [inline recipe/meta.yaml:45]:
```suggestion
```

We actually don't need the numpy >=1.18.5 pin, conda-build is building against the lowest supported numpy version from conda-forge (1.21) anyway 😅

**Comment by pavelzw** [comment]
@conda-forge-admin, please rerender

**Review by pavelzw — APPROVED** [review]

**Comment by kklein** [comment]
@pavelzw Would you agree that this failure looks unrelated to the change proposed by this PR?
https://dev.azure.com/conda-forge/feedstock-builds/_build/results?buildId=885856&view=logs&jobId=385a3385-ae8b-54d1-9abe-3c614f74cb73&j=385a3385-ae8b-54d1-9abe-3c614f74cb73&t=a5f9d343-149e-5621-6ab9-f0278009be04

**Comment by pavelzw** [comment]
yes this is just some flaky ci, i'll restart

**Pull request merged**